### ISSUE TITLE
Mistake in the calculation of probability density.

### ISSUE THREAD

**Issue by lszolnoki** [c0nrad/hydrogen#2]
From looking at the results and skimming the code it looks like that there is a mistake in calculating the probability density from the wave function.

https://github.com/c0nrad/hydrogen/blob/209ffe5cf14879a4679bc5368ddb0edb7c685838/hydrogen.cpp#L205

It should be |Phi|^2 = conj(Phi)*Phi, from the plot and from line of code above it looks more along the lines of Re[Phi^2], which is not the same. Since the resulting value is interpreted as a probability, the negative values from Re[Phi^2] are truncated.

For these quantum numbers Phi ~ exp(i m phi), |Phi|^2 supposed to have rotational symmetry along the z axis.

**Comment by c0nrad** [comment]
Wow, I'm so sorry I missed! I think you're right. I'm taking a peak now.

**Comment by c0nrad** [comment]
Just deployed the updated version. Thanks so much @lszolnoki! Does this version look better? https://blog.c0nrad.io/hydrogen/

**Comment by lszolnoki** [comment]
I looked at the deployed plots at https://blog.c0nrad.io/hydrogen/, unfortunately it still looks off. I'm looking at 422, and when you view from the top you can see whole quadrants missing. Remember, it is supposed to be have rotational symmetry.

I looked at your code changes and I didn't find any obvious mistake in your wave function calculation now, although I'm not familiar with ginac. In any case the phi-dependence should be fixed now, but I don't see it in the plots. Maybe it's just something banal like a discrepancy between your deployed version and the fixed version.

Where it gets hard to follow for me is the way you handle randomness. I have a suspicion of a missing Jacobian, but that would have an entirely different effect on the plots.

**Comment by c0nrad** [comment]
Wow. So the issue was I needed to swap theta/phi when using THREEJS.fromSphericalCoordinates. 
Didn't realize people used spherical coordinates differently.

The issue is fixed on my javascript version of this project: https://blog.c0nrad.io/hydrogenjs/#/
I'll fix this one in a bit. Thanks again @lszolnoki for bringing it up.

**Comment by lszolnoki** [comment]
Great! It looks ok now.
